_posts/2016-11-29-future.md

```diff
@@ -0,0 +1,215 @@
+
+### Take in a Dependency Graph / Operate Multifile
+
+Then we could integrate with other tools like Webpack better. This would allow cross file transformations or whole codebase optimizations. The main reason for this would be for the minifier (being able to remove properties based on checking the usage across the whole application) or for example providing errors for missing/invalid import/exports.
+
+### Parser Errors
+
+Better error messages from the parser like in [Compiler Errors for Humans](http://elm-lang.org/blog/compiler-errors-for-humans).
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">babel-eslint@7.1.1: now adds the code frame when there&#39;s a parser error! <a href="https://t.co/yoxRpGXq5E">pic.twitter.com/yoxRpGXq5E</a></p>&mdash; Henry Zhu (@left_pad) <a href="https://twitter.com/left_pad/status/799388723896946692">November 17, 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+> It is obvious that we all want to see helpful errors!
+
+- [babel/babylon#125](https://github.com/babel/babylon/pull/125): Instead of "Unexpected token" explain that `await` needs to be used in an async function.
+- Do better inference/guessing on user intention to prevent vague errors.
+
+### `babel-init`
+
+Basically a way to setup babel easier like create-react-app does.
+
+- Setup a `.babelrc` from scratch
+- Maybe reuse the `babel` npm package to default to use babel-preset-env
+
+### Run [tc39/test262](https://github.com/tc39/test262)
+
+Running official spec tests against Babel to make sure we comply to the spec or at least know when we don't.
+
+### Smoke Tests
+
+It would be useful to either have a reverse [Greenkeeper](https://greenkeeper.io) or to run tests with master branch of Babel so that we can catch any major regressions before any release. In theory we would want to take the biggest open source projects that use babel and run their tests.
+
+[motiz88/babel-smoke-tests](https://github.com/motiz88/babel-smoke-tests) is a start!
+
+### API Documentation / Community
+
+Meetups, tutorials, videos, open source workshops, etc.
+Docs should explain each feature and possible usages, improved links to books, articles, etc.
+
+- All plugins should have examples. Can also embed [RunKit](https://runkit.com/home) or the REPL.
+- Update Learn ES2015 to have ES2015+.
+- Add resources page
+- Update FAQ with common errors
+- Page for community plugins
+- babel-handbook, API docs, etc
+
+### Program Analysis
+
+- [Program Slicing](https://en.wikipedia.org/wiki/Program_slicing) with [slice-js](http://slides.com/kentcdodds/faster-javascript#/)
+- Semantic (AST aware) Grepping tool
+
+### `babel --debug`
+
+This command would print out all info (also when erroring). It would also include performance metrics on how long each plugin takes.
+
+### Syntax plugins:
+
+#### Pluggable Parser
+
```

Comment:
Feel free to (not required to) reference https://github.com/TheLarkInn/js-parser-discussions it has been initially discussed but this is something I see that could bring huge unity in the tooling land and interoperability. Initial discussions and analysis was made but I haven't gotten back to it in a bit since webpack is primary focus.